operator/README.md

```diff
@@ -1,2 +1,92 @@
+
+The PMEM-CSI operator is based on the CoreOS [operator-sdk](https://github.com/operator-framework/operator-sdk) tools and APIs.
+
+## PMEM-CSI Deployment custom resource definition(CRD)
+
+Current PMEM-CSI Deployment object supports below API:
+
+### Deployment
+
+|Field | type | Description |
+|---|---|---|
+| apiVersion | string  | `pmem-csi.intel.com/v1alpha1`|
+| kind | string | `Deployment`|
+| metaData | [ObjectMeta](https://git.k8s.io/community/contributors/devel/api-conventions.md#spec-and-status) | Standard objects metadata. |
+| spec | [DeploymentSpec](#deployment-spec) | Specification of the desired behavior of the deployment |
+
+### DeployemtSpec
+
+|Field | type | Description | Default Value |
+|---|---|---|---|
+| image | ImageInfo | PMEM-CSI driver container image information | `intel/pmem-csi-driver:canary` |
+| logLevel | integer | PMEM-CSI driver logging level | 3 |
+| controller | CotrollerDriver | configuration of PMEM-CSI driver running as controller | |
+| node | NodeDriver | configuration of PMEM-CSI driver running at compute node | |
+
+### ControllerDriver
+
+|Field | type | Description | Default Value |
+|---|---|---|---|
+| provisionerImager | ImageInfo | CSI provisioner container image information | `quay.io/k8scsi/csi-provisioner:v1.2.1` |
+| resources | [ResourceRequirements](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.12/#resourcerequirements-v1-core) | describes the compute resource requirements for controller pod |
+
+### NodeDriver
+
+|Field | type | Description | Default Value |
+|---|---|---|---|
+| deviceMode | string | Device management mode to use. Supports one of `lvm` or `direct` | `lvm`
+| registrarImage | ImageInfo | CSI provisioner container image information | `quay.io/k8scsi/csi-node-driver-registrar:v1.1.0` |
+| resources | [ResourceList](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.12/#resourcerequirements-v1-core) | |
+
+### ImageInfo
+
+|Field | type | Description | Default value |
+|---|---|--|---|
+| registry | string | Docker image registry |
+| name | string | Container image name |
+| tag | string | Container image tag |
+| pullPolicy | string | Image pull policy. either one of `Always`, `Never`, `IfNotPresent` | `IfNotPresent` |
+
+## Pre-requisites
+
+### Build operator image
+
+Build PMEM-CSI operator image using the provided [Dockerfile](https://github.com/intel/pmem-csi/tree/devel/operator/build/Dockerfile) and push the image to local docker registry.
+
+```sh
+docker build --pull -t pmem-csi-operator:canary  --label revision=$(VERSION) -f operator/build/Dockerfile .
+```
+
+### Deploy PMEM-CSI Deployment CRD
+
+```sh
+kubectl create -f https://raw.githubusercontent.com/avalluri/pmem-CSI/operator/operator/deploy/crds/pmem-csi.intel.com_deployments_crd.yaml
+```
+> FIXME: **Should this be handled with in operator**?
```

Comment:
Do you know how this is handled in other operators?

Reply:
The external-snapshotter used to create the CRD.

Reply:
> Do you know how this is handled in other operators?

I didn't check. But, my feeling is that the CRD is owned by this Operator and the end-user is no way concerned about altering this. So, I see no extra benefit of asking user to first install CRD yourself before starting Operator.

Reply:
I agree that letting the operator install and/or update the CRD seems better.

Reply:
Made needed changes to auto-install the CRD within operator: https://github.com/avalluri/pmem-CSI/commit/c574c4b9661c56cdba42446db9b27aa4c2285822